### DISCUSSION TITLE
Consent form front-end only

### DISCUSSION THREAD

**Discussion by robsonsobral** [ExpressionEngine/ExpressionEngine#3316] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

The LGPD, the Brazilian version of GDPR, [unlike this one](https://github.com/ExpressionEngine/ExpressionEngine/issues/454), doesn't require us to register submitted granting of consents.

Thus, Brazilian users are used to an instantaneous consent form, without the waiting for the request.

**Describe the solution you'd like**

I would like to have an option to make the consent form of anonymous users JavaScript only, without the need to wait for the server.

**Describe alternatives you've considered**

My other option is to replicate the functionality by myself, to the anonymous users.

**Teachability, Documentation, Adoption, Migration Strategy**
The current implementation has some backend moving parts.

From the [Consent Service](https://github.com/ExpressionEngine/ExpressionEngine/blob/1ece88cb14b85d06a55a751a5198032752f9cfe8/system/ee/ExpressionEngine/Service/Consent/Consent.php#L121):

```php
        if ($this->isAnonymous()) {
            $this->cookie[$request->getId()] = ['has_granted' => true, 'timestamp' => $this->now];
            $this->saveConsentCookie($this->cookie);
            $request->log(sprintf(lang('consent_granted_log_msg'), lang($via)));
            $this->updateConsentCache($request);
        } else {
```
```php
    /**
     * Encodes, signs, and saves the consent cookie
     *
     * @param array $consented_to An associative array of granted consents with the
     *   request's ID as the array key.
     */
    protected function saveConsentCookie(array $consented_to)
    {
        $payload = ee('Encrypt/Cookie')->signCookieData($consented_to);
        // 60 * 60 * 24 * 360 = 31104000; A (almost) year of seconds
        $this->input_delegate->set_cookie(self::COOKIE_NAME, $payload, 31104000);
        $this->session_delegate->set_cache(__CLASS__, 'cookie_data_' . $this->member_id, $consented_to);
    }
```

From [Cookie Encrypt service](https://github.com/ExpressionEngine/ExpressionEngine/blob/1ece88cb14b85d06a55a751a5198032752f9cfe8/system/ee/ExpressionEngine/Service/Encrypt/Cookie.php#L54):

```php
    public function signCookieData($data)
    {
        $payload = json_encode($data, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);

        return $payload . $this->generateHashForCookieData($payload);
    }
```

 I don't know which is the best path to take and make it retrocompatible, but I would like to help implement it, if considered.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@robsonsobral so what exactly do you want to achieve here instead? 

(you can just hide the form upon AJAX submissions without waiting for the response)
